extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/ConnectionManager.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.relational.jdbc;
+
+import javax.sql.DataSource;
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConnectionManager {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConnectionManager.class);
+
+  private DataSource dataSource;
+
+  public ConnectionManager(RelationalJdbcConfiguration jdbcConfiguration) {
+    this.dataSource = initializeDataSource(jdbcConfiguration);
+  }
+
+  private BasicDataSource initializeDataSource(RelationalJdbcConfiguration jdbcConfiguration) {
+    BasicDataSource bds = new BasicDataSource();
```

Comment:
Why not use Quarkus infrastructure for managing the DataSource?

Example: https://github.com/projectnessie/nessie/blob/54f983c0e611daa77686060cc2be35a40814ad57/servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/storage/JdbcBackendBuilder.java#L121

That part will be present only in the server (i.e. future code).

Unit tests can, of course, use DBCP locally. WDYT?

Reply:
my understanding was that we need to use the jdbc via dbcp2 pooling even for prod https://github.com/apache/polaris/issues/649#issue-2779689169, are we ok now to use datasources managed by Quarkus. I am fine with either 

as based on the these configs we should be good : https://quarkus.io/guides/datasource#jdbc-configuration

Reply:
#649 might be suggesting a possible way to implement... I do not think it is the requited way :) @jbonofre WDYT?

Reply:
To clarify my proposed design:
* This module will provide JDBC persistence code
* It will receive a `DataSource` as a parameter at construction time
* The Quarkus runtime can provide the `DataSource` and manager related configuration automatically.
* If the module is reused somewhere else, the place where it is reused (e.g. in tests) will have to manage the `DataSource` by whatever means are appropriate there, which might as well be DBCP.

Reply:
Agree with @dimas-b - this must use the Quarkus infrastructure within `polaris-quarkus-*`, not a custom `DataSource`.

Reply:
sounds good, i will incorporate this feedback, a some follow-up questions : 
 I am assuming the DBOperations class which does (insert / write / transactions) gets a DS as constructor param which will injected via JdbcMetaStoreManagerFactory when its creating the `PolarisJDBCBasePersistenceImpl` ? this package will not have the config parsing logic etc. If yes how does Quarkus Admin / Server share this code of passing config ? 
if we want some sharing should i incorporate the config parsing such as [this](https://github.com/projectnessie/nessie/blob/54f983c0e611daa77686060cc2be35a40814ad57/servers/quarkus-common/src/main/java/org/projectnessie/quarkus/providers/storage/JdbcBackendBuilder.java#L41)

Reply:
please have a look, incorporated this feedback !